app/org/sagebionetworks/bridge/dynamodb/DynamoUploadDao.java

```diff
@@ -218,12 +218,14 @@ public Upload getUpload(@Nonnull String uploadId) {
         if (offsetKey != null) {
             // load table again to get the one last evaluated upload
             DynamoUpload2 retLastEvaluatedUpload = mapper.load(DynamoUpload2.class, offsetKey);
-
-            Map<String,AttributeValue> map = new HashMap<>();
-            map.put(UPLOAD_ID, new AttributeValue().withS(offsetKey));
-            map.put(REQUESTED_ON, new AttributeValue().withN(String.valueOf(retLastEvaluatedUpload.getRequestedOn())));
-            map.put(STUDY_ID, new AttributeValue().withS(studyId.getIdentifier()));
-            query.withExclusiveStartKey(map);
+            
+            if (retLastEvaluatedUpload != null) {
+                Map<String,AttributeValue> map = new HashMap<>();
+                map.put(UPLOAD_ID, new AttributeValue().withS(offsetKey));
+                map.put(REQUESTED_ON, new AttributeValue().withN(String.valueOf(retLastEvaluatedUpload.getRequestedOn())));
+                map.put(STUDY_ID, new AttributeValue().withS(studyId.getIdentifier()));
+                query.withExclusiveStartKey(map);
+            }
```

Comment:
What is the effect of adding this if block? Do we need a unit test to test this case?